### ISSUE TITLE
When joining server stuck on "Server start command issued. Please wait... 0%"

### ISSUE THREAD

**Issue by GabrielSerealia** [gekware/minecraft-server-hibernation#133]
When I join the server I indefinitely get the status message "Server start command issued. Please wait... 0%". I then have to go back to the server list and reconnect. Is this a known issue with the latest version of Minecraft/PaperMC? It's also possible it may be a configuration issue.

![issue](https://i.serealia.ca/2022/02/javaw_1dGPe0q_2022-02-23_03-10-23.png)

config.json
```JSON
{
  "Server": {
    "Folder": "/home/container",
    "FileName": "server.jar",
    "Protocol": "757",
    "Version": "Paper 1.18.1"
  },
  "Commands": {
    "StartServer": "java -Xms5G -Xmx5G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -jar server.jar nogui",
    "StopServer": "stop",
    "StopServerForce": "stop"
  },
  "Msh": {
    "CheckForUpdates": false,
    "Debug": false,
    "HibernationInfo": "                   \u0026fserver status:\n                   \u0026b\u0026lHIBERNATING",
    "Port": "25565",
    "StartingInfo": "                   \u0026fserver status:\n                    \u00266\u0026lWARMING UP",
    "TimeBeforeStoppingEmptyServer": 60
  }
}
```

**Comment by gekigek99** [comment]
I think it's a known issue but I'm not sure.

Can you specify what msh version are you using?

After setting loglevel to 3 in config can you share also the log?

**Comment by GabrielSerealia** [comment]
Yes, it does let me connect after doing that.
I was just wondering if the status message would update as the server boots.

I am using 2.4.0
I've updated to 2.4.4 but it's doing the same thing, though I do like how clean everything looks and the addition of the exit command instead of ^C
I'm running paper but it's actually been running fine with it other than the "Server start command issued. Please wait... 0%" staying at 0%

Thanks for making this application btw it really saves me a lot of resources as I've had this world running almost 24/7 for a small group of friends since 2010 :)

**Comment by gekigek99** [comment]
> I was just wondering if the please wait status message would update as the server boots.

it should teoretically, but probably the paper server stays at `0%` for some seconds and then in half a second it loads to 100%. To see it changing you should try to join exacly when the percentage changes. (the `wait status message` is static, to see it change you have to reload the page)

> Do you think you could also add back the console line that prints when the process finishes loading correctly please?
Previously it was "waiting for clients to connect..."

it's still there, you have to set loglevel to 3 instead of 1 tho ahah ;) now that i think about it yeah i should probably set it to loglevel 1 again (did it in 4cb555217295cc2ad5dd0f26f7d92c34629eaffb)

> And I just wanted to confirm if the "msh exit" command gracefully exists like ^C in 2.4.0
(After some testing the exit command doesn't appear to perform a graceful shutdown of the server, using ^C appears to gracefully shutdown)

you are completely right I'll fix that when I have a bit more time... would you mind opening an issue for that? just copy what you just wrote

> Thanks for making this application btw it really saves me a lot of resources as I've had this world running almost 24/7 for a small group of friends since 2010 :)

I'm really happy I can help you! if you like this rep please consider giving it a star

**Comment by GabrielSerealia** [comment]
> it should teoretically, but probably the paper server stays at 0% for some seconds and then in half a second it loads to 100%. To see it changing you should try to join exacly when the percentage changes. (the wait status message is static, to see it change you have to reload the page)

Ah, okay so my initial ticket was because I thought it was dynamic lmao

> it's still there, you have to set loglevel to 3 instead of 1 tho ahah ;) now that i think about it yeah i should probably set it to loglevel 1 again (did it in https://github.com/gekware/minecraft-server-hibernation/commit/4cb555217295cc2ad5dd0f26f7d92c34629eaffb)

In regards to this I use Pterodactyl Panel to manage the server and it checks for specific lines in the console to determine if the docker container booted correctly. I used to use the "waiting for clients to connect..." message from MSH to determine if it finished loading. (I mean the loading configs line should be good enough)

> you are completely right I'll fix that when I have a bit more time... would you mind opening an issue for that? just copy what you just wrote

Sure thing

Do you take donations btw?

**Comment by gekigek99** [comment]
> Ah, okay so my initial ticket was because I thought it was dynamic lmao

lol ;)

> I used to use the "waiting for clients to connect..." message from MSH to determine if it finished loading. (I mean the loading configs line should be good enough)

you can set the loglevel to 3 in config and the string should be there again

Do you take donations btw?

yeah sure [here](https://www.buymeacoffee.com/gekigek99)

p.s. there is also a [discord server](https://discord.com/invite/mBSnEypcYC) to ask for help and chill ;)

**Comment by gekigek99** [comment]
This issue was indirectly solved with 1c21a5ee5c473f61c19d17d2a3e9c19ccd9a0dd5

You can set AllowSuspend to true to avoid player waiting time while joining a frozen server
